### DISCUSSION TITLE
Linear Advance firmware help

### DISCUSSION THREAD

**Discussion by brunohirt** [mriscoc/Ender3V2S1#392] [Q&A] [answered]
Hello everyone!

I'm an old Ender 3 Pro user, and recently upgraded to a S1 Pro (big mistake, now I know), but found very difficult to print good quality prints fast, like I used to do with my old printer. So I found mriscoc's Pro Firmware and it's benefits over the original Creality Firmware, so I got a DWIN display, upgraded the firmware like the tutorials and I'm back printing, but the print's bulge on corners continue at higher speeds, and I would definitely like to enable Linear Advance on my firmware, but, being really honest, I tried following the tutorials, using Visual Code and Python and man, sorry, it may be easy, but not for present me.

If someone could help me making the firmware I need, it would be awesome, because I got really lost trying to make it myself and got really nowhere.

I know that some functions must be disabled in order for the F4 chip memory to be able to run LA, so I'm open to suggestions on what to do.

- Ender 3 S1 Pro - F4 SoC

Thank you very much, guys!

**Comment by brunohirt** [comment, marked as answer]
Found out how to use the firmware configurator, so for F4 to have LA enabled, I must disable UBL, is that right?

**Reply by mriscoc** [reply]
To make space for LA the following can be disabled:
```
BAUD_RATE_GCODE
PRINTCOUNTER
POWER_LOSS_RECOVERY
BINARY_FILE_TRANSFER
```
and SD card in read only mode.

**Reply by brunohirt** [reply]
Thanks! I disabled ARC for now just to try it out, and it worked, will try those features soon. So if you want to close this discussion as answered, go ahead!

**Reply by nwpayton** [reply]
do you stilll have the BIN? id love to have that im having trouble with configurator

**Reply by sgtpepperaut** [reply]
hey THAT BIN would be much appreciated :)

**Reply by sgtpepperaut** [reply]
> and SD card in read only mode.

do you mean physically moving the slider on the SD?

**Reply by mriscoc** [reply]
{
    "op": "Enable",
    "searchfor": "**SDCARD_READONLY**"
  },
